src/bin/genemichaels.rs

```diff
@@ -1,177 +1,213 @@
+        default_value_t = match FormatConfig::default().comment_width {
+            Some(x) => Offable::On(x),
+            None => Offable::Off,
+        },
+    )]
+    comment_length: Offable<usize>,
+    #[arg(
+        long,
+        help = "Problems formatting comments are fatal; disable with `false`",
+        default_value_t = match FormatConfig::default().comment_errors_fatal {
+            true => Offable::On(On),
+            false => Offable::Off,
+        },
+    )]
+    comment_errors_fatal: Offable<On>,
+}
 
-    fn process(config: &FormatConfig, source: &str) -> Result<String> {
-        let res = format_str(source, config)?;
-        if !res.lost_comments.is_empty() {
+fn process(config: &FormatConfig, source: &str) -> Result<String> {
+    let res = format_str(source, config)?;
+    if !res.lost_comments.is_empty() {
+        return Err(
+            anyhow!(
+                "The following comments were missed during formatting: {:?}",
+                res.lost_comments.values().flatten().collect::<Vec<&Comment>>()
+            ),
+        );
+    }
+    match syn::parse_str::<File>(&res.rendered) {
+        Ok(_) => { },
+        Err(e) => {
+            eprintln!("{}", res.rendered);
             return Err(
                 anyhow!(
-                    "The following comments were missed during formatting: {:?}",
-                    res.lost_comments.values().flatten().collect::<Vec<&Comment>>()
+                    "Rendered document couldn't be re-parsed in verification step: {}:{}: {}",
+                    e.span().start().line,
+                    e.span().start().column,
+                    e
                 ),
             );
+        },
+    };
+    Ok(res.rendered)
+}
+
+fn skip(src: &str) -> bool {
+    src.lines().take(5).any(|l| l.contains("`nogenemichaels`"))
+}
+
+fn process_workspace_file(config: FormatConfig, file_path: PathBuf) {
+    let res = || -> Result<()> {
+        let source = fs::read_to_string(file_path.clone())?;
+        if skip(&source) {
+            print_skipping_text();
+            eprintln!("{}", &file_path.to_string_lossy());
+            return Result::Ok(());
         }
-        match syn::parse_str::<File>(&res.rendered) {
-            Ok(_) => { },
-            Err(e) => {
-                eprintln!("{}", res.rendered);
-                return Err(
-                    anyhow!(
-                        "Rendered document couldn't be re-parsed in verification step: {}:{}: {}",
-                        e.span().start().line,
-                        e.span().start().column,
-                        e
-                    ),
-                );
-            },
-        };
-        Ok(res.rendered)
+        fs::write(&file_path, process(&config, &source)?.as_bytes())?;
+        Ok(())
+    };
+    match res() {
+        Ok(_) => {
+            // bold green
+            eprint!("\x1B[1;32m");
```

Comment:
I _think_ splitting up the print like this could allow multiple threads to be interleaved leading to strangely formatted output. Could these be combined into one eprintln maybe? (or doing the explicit lock stderr thing)

Reply:
Good call.